R/data-importer-configuration.R

```diff
@@ -239,28 +238,31 @@ DataImporterConfiguration <- R6::R6Class(
     }
   ),
   public = list(
-    #' @description
-    #' Initialize a new instance of the class
-    #' @param configurationFilePath Path to the XML file with stored configuration
-    #' (e.g. create in PK-Sim or MoBi).
+    #' @param ref Reference to .NET DataImporterConfiguration object
     #' If `NULL` (default), an empty configuration with columns "Time" and
     #' "Measurement" is created.
+    #' @description
+    #' Initialize a new instance of the class
     #' @return A new `DataImporterConfiguration` object.
-    initialize = function(configurationFilePath = NULL) {
+    initialize = function(ref = NULL) {
       importerTask <- getNetTask("DataImporterTask")
-
-      if (is.null(configurationFilePath)) {
+      if (is.null(ref)) {
         ref <- rClr::clrCall(importerTask, "CreateConfiguration")
-      } else {
-        validateIsString(configurationFilePath)
-        ref <- rClr::clrCall(importerTask, "GetConfiguration", configurationFilePath)
       }
       super$initialize(ref)
       private$.dataImporterTask <- importerTask
 
-      private$.timeColumn <- rClr::clrCall(importerTask, "GetTime", ref)
-      private$.measurementColumn <- rClr::clrCall(importerTask, "GetMeasurement", ref)
-      private$.errorColumn <- rClr::clrCall(importerTask, "GetError", ref)
+      # set timeColumn dimension and unit to default ("Time" and "h") if it is
+      # not set properyl in the .NET object. This could happen when creating
+      # a configuration for a xls-sheet and the unit could not be recognized.
+      # Because the user cannot set the Dimension if time values, this must be
+      # done during initialization phase.
+      if (self$timeUnit == "?") {
```

Comment:
Probably only relevant when "guessing" the configuration for a sheet, but better having it in the constructor.